libraries/fabric-shim/lib/server.js

```diff
@@ -59,7 +59,15 @@ class ChaincodeServer {
         }
 
         // Create GRPC Server and register RPC handler
-        this._server = new grpc.Server();
+        this._server = new grpc.Server({
+            'grpc.max_send_message_length': serverOpts['grpc.max_send_message_length'],
+            'grpc.max_receive_message_length': serverOpts['grpc.max_receive_message_length'],
+            'grpc.keepalive_time_ms': serverOpts['grpc.keepalive_time_ms'],
+            'grpc.http2.min_time_between_pings_ms': serverOpts['grpc.http2.min_time_between_pings_ms'],
+            'grpc.keepalive_timeout_ms': serverOpts['grpc.keepalive_timeout_ms'],
+            'grpc.http2.max_pings_without_data': serverOpts['grpc.http2.max_pings_without_data'],
+            'grpc.keepalive_permit_without_calls': serverOpts['grpc.keepalive_permit_without_calls'],
+        });
```

Comment:
There are more options that can be passed to the `grpc.Server` constructor, and that list of options might change over time. Maybe just passing through any properties beginning with "grpc." would cover the possibilities more simply?
```suggestion
        const grpcOptions = Object.fromEntries(
            Object.entries(serverOpts).filter(([key]) => key.startsWith('grpc.'))
        );
        this._server = new grpc.Server(grpcOptions);
 ```

Reply:
Hello Mark, I chose to be explicit on the entries for the object because those were the ones specified on fabric-shim/cmds/serverCommand.js under validOptions. I treated this as a bug, since any one of those weren't being passed to the gRPC Server.

I agree that there is discrepancies in the index.d.ts in regards to this ChaincodeServerOpts interface and the what serverCommand/startCommand.js are implying with their validOptions.

Reply:
To keep the code simple and the CLI command more easily extended to accommodate additional gRPC options, I think I would still be tempted to pass any property that starts with `grpc.` on to the gRPC Server, and let the CLI command worry about explicitly defining the options that the CLI command accepts. Whichever approach you prefer is OK with me though. Just the JSDoc and TypeScript type definitions need to match the implementation.